internal/api/handler_test.go

```diff
@@ -0,0 +1,306 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/JakeHumphries/gymshark-hacker-news/internal/models"
+	"github.com/labstack/echo/v4"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type MockReader struct {
+	mock.Mock
+}
+
+func (m *MockReader) GetAllItems(ctx context.Context) ([]models.Item, error) {
+	args := m.Called()
+
+	itemArg, ok := args.Get(0).(*models.Item)
```

Comment:
You should be able to use `args.Get(0).([]models.Item])` ([see here](https://github.com/gymshark/realm-migration-service/blob/main/internal/realmlegacy/mock.go#L13))